pkg/provisioner/ironic/ironicservices.go

```diff
@@ -0,0 +1,103 @@
+package ironic
+
+import (
+	"context"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/baremetal/v1/drivers"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+func waitForIronicServices(client *gophercloud.ServiceClient, inspector *gophercloud.ServiceClient, timeout int) bool {
+
+	return waitForIronic(client, timeout) &&
+		waitForInspector(inspector, timeout)
+}
+
+func waitForIronic(ironic *gophercloud.ServiceClient, timeout int) bool {
+	// Let's poll the API until it's up, or times out.
+	duration := time.Duration(timeout) * time.Second
```

Comment:
This calculation is being done everywhere. Why not make timeout a Duration type? Or move the context handling to waitForIronicServices()?